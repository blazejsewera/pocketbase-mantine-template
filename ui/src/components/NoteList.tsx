import { Card, Text } from '@mantine/core'
import { Note } from '../domain/Note'

interface NoteCardProps {
  note: Note
}

const NoteCard = ({ note }: NoteCardProps) => (
  <Card shadow="sm" p="lg" radius="md" withBorder>
    <Text>{note.title}</Text>
    <Text>{note.body}</Text>
  </Card>
)

export interface NoteListProps {
  notes: Note[]
}

export const NoteList = ({ notes }: NoteListProps) => (
  <div>{notes.length ? notes.map(note => <NoteCard key={note.title} {...{ note }} />) : <Text>Empty</Text>}</div>
)
