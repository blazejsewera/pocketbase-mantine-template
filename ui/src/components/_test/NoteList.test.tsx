import { snap } from '../../util/test/snapshot'
import { NoteList } from '../NoteList'

it('renders empty list properly', () => {
  const s = snap(<NoteList notes={[]} />)
  expect(s).toMatchSnapshot()
})

it('renders list with two elements properly', () => {
  const s = snap(
    <NoteList
      notes={[
        { title: '1', body: '1' },
        { title: '2', body: '2' },
      ]}
    />,
  )
  expect(s).toMatchSnapshot()
})
