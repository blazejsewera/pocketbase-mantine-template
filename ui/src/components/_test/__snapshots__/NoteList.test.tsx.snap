// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty list properly 1`] = `
<div>
  <div
    className="mantine-Text-root mantine-1pbxw0k"
  >
    Empty
  </div>
</div>
`;

exports[`renders list with two elements properly 1`] = `
<div>
  <div
    className="mantine-Paper-root mantine-Card-root mantine-ergsfy"
  >
    <div
      className="mantine-Text-root mantine-1pbxw0k"
    >
      1
    </div>
    <div
      className="mantine-Text-root mantine-1pbxw0k"
    >
      1
    </div>
  </div>
  <div
    className="mantine-Paper-root mantine-Card-root mantine-ergsfy"
  >
    <div
      className="mantine-Text-root mantine-1pbxw0k"
    >
      2
    </div>
    <div
      className="mantine-Text-root mantine-1pbxw0k"
    >
      2
    </div>
  </div>
</div>
`;
